src/olympia/activity/utils.py

```diff
@@ -0,0 +1,210 @@
+import base64
+import datetime
+import logging
+import quopri
+import re
+import urllib2
+from email import message_from_string
+from email.utils import parseaddr
+
+from django.conf import settings
+from django.template import Context, loader
+
+from email_reply_parser import EmailReplyParser
+
+from olympia import amo
+from olympia.access import acl
+from olympia.activity.models import ActivityLogToken
+from olympia.amo.helpers import absolutify
+from olympia.amo.urlresolvers import reverse
+from olympia.amo.utils import send_mail
+from olympia.devhub.models import ActivityLog
+
+log = logging.getLogger('z.amo.activity')
+
+# Prefix of the reply to address in devcomm emails.
+REPLY_TO_PREFIX = 'reviewreply+'
+
+
+class ActivityEmailParser(object):
+    """Utility to parse email replies."""
+    address_prefix = REPLY_TO_PREFIX
+
+    def __init__(self, email_text):
+        """Decode base64 email and turn it into a Django email object."""
+        try:
+            email_text = base64.standard_b64decode(
+                urllib2.unquote(email_text.rstrip()))
+        except TypeError:
+            # Corrupt or invalid base 64.
+            self.decode_error = True
+            log.info('Decoding error for ActivityEmailParser')
+            return
+
+        self.email = message_from_string(email_text)
+
+        payload = self.email.get_payload()
+        if isinstance(payload, list):
+            # If multipart, get the plain text part.
+            for part in payload:
+                # Nested multipart. Go deeper.
+                if part.get_content_type() == 'multipart/alternative':
+                    payload = part.get_payload()
+                    for part in payload:
+                        if part.get_content_type() == 'text/plain':
+                            # Found the plain text part.
+                            payload = part.get_payload()
+                            break
+
+                if part.get_content_type() == 'text/plain':
+                    # Found the plain text part.
+                    payload = part.get_payload()
+                    break
+
+        # Decode quoted-printable data and remove non-breaking spaces.
+        payload = (quopri.decodestring(payload)
+                         .replace('\xc2\xa0', ' '))
+        payload = self.extra_email_reply_parse(payload)
```

Comment:
Hmm, I'm assuming that the forwarded mail is the original one without special treatment from Postfix or so? If so, the world lives in a weird place regarding encodings and we might want to use `charded` to properly decode things. I wrote a little helper some time ago for an experimental email fetching thingy, https://github.com/mailme/mailme/blob/master/src/mailme/utils/parser.py#L107 which might be helpful, or not.